src/test/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsBaseTest.java

```diff
@@ -277,4 +287,59 @@ void testTableStream(MtAmazonDynamoDbBase mtDynamoDb) {
         }
     }
 
+    /**
+     * Verifies that reading virtual tables with streams disabled doesn't return records.
+     */
+    @ParameterizedTest
+    @ArgumentsSource(Args.class)
+    void testDisabledStreams(MtAmazonDynamoDbBase mtDynamoDb) {
+        final MtAmazonDynamoDbStreams mtDynamoDbStreams = MtAmazonDynamoDbStreams.createFromDynamo(mtDynamoDb,
+            AmazonDynamoDbLocal.getAmazonDynamoDbStreamsLocal());
+        try {
+            String tenant = TENANTS[0];
+
+            // create table with streams enabled
+            String tableWithStreamsEnabled = TENANT_TABLE_NAME + "_streams_enabled";
+            CreateTableRequest createTableRequestStreamsEnabled =
+                newCreateTableRequest(tableWithStreamsEnabled)
+                    .withStreamSpecification(new StreamSpecification()
+                        .withStreamEnabled(true)
+                        .withStreamViewType(NEW_AND_OLD_IMAGES));
+            MT_CONTEXT.withContext(tenant, () -> mtDynamoDb.createTable(createTableRequestStreamsEnabled));
+
+            // create table with streams disabled
+            String tableWithStreamsDisabled = TENANT_TABLE_NAME + "_streams_disabled";
+            CreateTableRequest createTableRequestStreamsDisabled =
+                newCreateTableRequest(tableWithStreamsDisabled)
+                    .withStreamSpecification(new StreamSpecification()
+                        .withStreamEnabled(false)
+                    );
+            MT_CONTEXT.withContext(tenant, () -> mtDynamoDb.createTable(createTableRequestStreamsDisabled));
+
+            // put an item in each table
+            final MtRecord streamsEnabledRecord = putTestItem(mtDynamoDb, tableWithStreamsEnabled, tenant, 1);
+            putTestItem(mtDynamoDb, tableWithStreamsDisabled, tenant, 2);
+
+            // verify that the record from the stream enabled table is returned
+            MT_CONTEXT.withContext(tenant, () -> {
+                String streamArn = mtDynamoDb.describeTable(tableWithStreamsEnabled).getTable().getLatestStreamArn();
+                assertNotNull(streamArn);
+                String iterator = getShardIterator(mtDynamoDbStreams, streamArn).get();
+                assertGetRecords(mtDynamoDbStreams, iterator, streamsEnabledRecord);
+            });
+
+            // verify that either the streamArn is null(for ByTable or ByAccount) or the iterator returns no records
+            MT_CONTEXT.withContext(tenant, () -> {
+                String streamArn = mtDynamoDb.describeTable(tableWithStreamsDisabled).getTable().getLatestStreamArn();
+                if (streamArn != null) {
+                    // for byIndex, the streamArn will not be null, but it should return no records
+                    String iterator = getShardIterator(mtDynamoDbStreams, streamArn).get();
```

Comment:
So this test currently fails when it tests ByIndex, because it gets records from the other table that it is co-located whose virtual table has streams enabled.  Seems like the test is right and the implementation is wrong no?

Reply:
Agree, that's a bug. The implementation is only filtering out records for other tenants, but not records for other virtual tables. We will probably need to give each virtual table it's own virtual arn, so that getShardIterator/getRecords knows what to return.

Reply:
https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html